sunbeam-python/sunbeam/jobs/manifest.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import yaml
+from pydantic import Field
+from pydantic.dataclasses import dataclass
+
+from sunbeam.clusterd.client import Client as clusterClient
+from sunbeam.jobs.common import BaseStep, Result, ResultType, Status
+
+LOG = logging.getLogger(__name__)
+EMPTY_MANIFEST = """charms: {}
+terraform-plans: {}
+"""
+VALID_TERRAFORM_PLANS = [
```

Comment:
I don't think that's the right way, implemented like this, we have to know all the plans sunbeam is using, on every plugins (even external if we want to be able to override them)

Reply:
Modified so that the plugin can specify the keys and used as part of validation.